packages/commonwealth/client/scripts/views/layout.tsx

```diff
@@ -116,24 +117,38 @@ class LayoutComponent extends ClassComponent<LayoutAttrs> {
       this.loadingScope = scope;
       if (scopeMatchesChain) {
         this.deferred = deferChain;
-        selectChain(scopeMatchesChain, deferChain).then((response) => {
-          if (!deferChain && response) {
-            initChain().then(() => {
+        this.loadingChain = true;
+        selectChain(scopeMatchesChain, deferChain)
+          .then((response) => {
+            if (!deferChain && response) {
+              initChain().then(() => {
+                this.redraw();
+              });
+            } else {
               this.redraw();
-            });
-          } else {
-            this.redraw();
-          }
-        });
+            }
+          })
+          .finally(() => (this.loadingChain = false));
         return <LoadingLayout />;
       }
     }
 
-    if (scope && this.deferred && !deferChain) {
+    if (
+      (scope && this.deferred && !deferChain) ||
+      (scope &&
+        scope === app.activeChainId() &&
+        scope !== this.loadingScope &&
+        app.chain &&
+        !app.chain.loaded &&
+        !deferChain)
```

Comment:
it works for me now => both direct access and with click.
Having said that, I have no idea what could be a side effect and I have no idea what that condition means. So if we want to stick with this solution, probably the condition should be moved to some variable and described with comment why we need to do this to make sure that we dont forget that in 3 days.

Reply:
cc @jnaviask please add more insights and/or accept the PR

Reply:
I am extremely hesitant to modify conditions in layout.tsx, as it impacts literally every page on the site. We should be very very clear + document what changes to this file do + why we are making them. This logic is not at all clear to me at first glance.

Reply:
The issue here was related to chain load deferring. To reproduce
1. Visit a page that deferres chain load (`/dydx/discussions`) - when the page is loaded we have `scope`, `app.activeChainId()`, and `this.loadingScope` defined all with `dydx` as value
2. Click on the new proposal link (must do it via link click) - on this step we have `scope` and `app.activeChainId()` defined as `dydx` but the `this.loadingScope` is undefined and also we need to load the chain on this page

What the new change, we are checking for `scope` and `app.activeChainId()` to be defined and match and the `loadingScope` to not exist, if this happens we assume we have an active chain, next we check if the chain was loaded with `app.chain && !app.chain.loaded` and if we need to defer with `deferChain`. If all good then we assume the active page needs the chain to be loaded and do that.

Reply:
This [commit](https://github.com/hicommonwealth/commonwealth/pull/3476/commits/0fd649759a46b285b9b313b5fa43601498374711) simplifies it a bit, but still I think the layout components has too many cross-cutting concerns.